### PR TITLE
Speedrun Timer - First Commit

### PR DESCRIPTION
**Pull request by CRGorman**

### CODE REVIEW

File: SpeedrunTimer/data/tables/speedrun-timer-sct.tbm [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+$On Load Complete:
+[
+    if (mn == nil) then
+        return
+    end
+    if (mn.getMissionFilename():upper() == "SM3-10A") or --SM3-10A is the FSPort Credits Mission
+        (mn.getMissionFilename():upper() == "BTM-01") or --Training One in the FSPort
+        (mn.getMissionFilename():upper() == "BTM-02") or --Training Two in the FSPort
+        (mn.getMissionFilename():upper() == "BTM-03") or --Training Three in the FSPort
```
Comment (Goober5000) [inline SpeedrunTimer/data/tables/speedrun-timer-sct.tbm]: There are two additional training missions in the FSPort, BTM-04 and BTM-05.  Just wanted to double check that they were intentionally omitted.

File: SpeedrunTimer/data/tables/speedrun-timer-sct.tbm [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        self.RunningStatus = false
+        ba.println("Speedrun Timer: StopTimer called - Last Segment Time: " .. self.CurrentSegmentTime)
+        self.CurrentSegmentTime = 0
+        self:PauseLiveSplit();
+    end
+
+    function SpeedrunTimer:PauseLiveSplit() --Send Pause to LiveSplit, and split if its on a tracked segment.
+        ba.println("Speedrun Timer: PauseLiveSplit called - Mission Filename: " .. mn.getMissionFilename())
+        if not(mn.isTraining() or --Training Missions
+         mn.getMissionFilename():upper() == "SM1-01" or --Start of the game.
```
Comment (Goober5000) [inline SpeedrunTimer/data/tables/speedrun-timer-sct.tbm]: Does the FSPort mission ("SM1-01A") need to be checked here too?

File: SpeedrunTimer/data/tables/speedrun-timer-sct.tbm [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        ba.println("Speedrun Timer: Paused, Loading Screen")
+    end
+]
+
+$On Load Complete:
+[
+    if (mn == nil) then
+        return
+    end
+    if (mn.getMissionFilename():upper() == "SM3-10A") or --SM3-10A is the FSPort Credits Mission
```
Comment (Goober5000) [inline SpeedrunTimer/data/tables/speedrun-timer-sct.tbm]: You might consider doing something like
```
local filename = mn.getMissionFilename():upper()
```
and then checking `filename` instead

**Review by Goober5000 — APPROVED** [review]
A few things I noticed; feel free to take them or leave them.

File: SpeedrunTimer/data/tables/speedrun-timer-sct.tbm (hunk 1 above)
Comment (CRGorman) [reply SpeedrunTimer/data/tables/speedrun-timer-sct.tbm]: Intentional. Timer starts on Commit of the first proper mission, so the timer does not need to start until then.

File: SpeedrunTimer/data/tables/speedrun-timer-sct.tbm (hunk 2 above)
Comment (CRGorman) [reply SpeedrunTimer/data/tables/speedrun-timer-sct.tbm]: Nope. Since the timer isn't running yet.

**Comment by CRGorman** [comment]
I need to work out/finish an issue with making sure the end of the run is marked as finished.

File: SpeedrunTimer/data/tables/speedrun-timer-sct.tbm [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            self.LSPipe:write "split\n" --Split time.
+            self.LSPipe:flush()
+        end
+        self.LSPipe:write "pausegametime\n" --Stop the clock for the loading screen.
+        self.LSPipe:flush()
+        self.CurrentMission = mn.getMissionFilename()
+    end
+
+    function SpeedrunTimer:StartTimer()
+        if(ca == nil) then
```
Comment (Goober5000) [inline SpeedrunTimer/data/tables/speedrun-timer-sct.tbm]: "ca" is a library, so I don't think it is ever nil.  There is a `mn.isInCampaign()` function if that's what you want.

File: SpeedrunTimer/data/tables/speedrun-timer-sct.tbm [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+[
+    if SpeedrunTimer.RunningStatus == true then
+        SpeedrunTimer:StopTimer()
+        ba.println("Speedrun Timer: Paused, Loading Screen")
+    end
+]
+
+$On Load Complete:
+[
+    if (mn == nil) then
```
Comment (Goober5000) [inline SpeedrunTimer/data/tables/speedrun-timer-sct.tbm]: "mn" is also a library, so I don't think it is nil either.  You might want `mn.isInMission()`

**Review by Goober5000 — COMMENTED** [review]
Additional comments

**Review by Goober5000 — APPROVED** [review]
Great stuff 👍

**Pull request merged**